load.php

```diff
@@ -36,23 +36,28 @@
 
 // Testing constants to switch to test db/redis/elastic.
 add_action( 'altis.modules.init', function() {
-	// Detect requests from Codeception/phpunit, primarely to use the test database.
+	// Detect requests from Codeception/phpunit, primarily to use the test database.
 	$is_test_request = (
 		isset( $_SERVER['HTTP_X_TEST_REQUEST'] )
 		|| ( isset( $_SERVER['HTTP_USER_AGENT'] ) && strpos( $_SERVER['HTTP_USER_AGENT'], 'wp-browser' ) !== false )
 		|| getenv( 'WPBROWSER_HOST_REQUEST' )
+		// Ensure CLI commands have the test context.
+		|| ( defined( 'WP_CLI' ) && WP_CLI && file_exists( Altis\ROOT_DIR . '/vendor/.test-running' ) )
+		|| ( class_exists( 'HM\\Cavalcade\\Runner\\Runner' ) && file_exists( Altis\ROOT_DIR . '/vendor/.test-running' ) )
```

Comment:
`WPBROWSER_HOST_REQUEST` should be defined in cli tests as per `wp-browser` docs, was that not the case ?
Only concern here is the `.tests-running` file not being cleared in case of any fatal error or something.

Reply:
Actually, I think it is possible for the user to halt execution of the command, so the file would not be cleared then. For the `test` database, it is being overridden on each execution, so cleanup isn't crucial, but for that file,  it might affect how the site behaves normally if the file is not removed due to the user halting it mid-way.

Reply:
From https://www.php.net/manual/en/function.register-shutdown-function.php

> Shutdown functions will not be executed if the process is killed with a SIGTERM or SIGKILL signal. While you cannot intercept a SIGKILL, you can use pcntl_signal() to install a handler for a SIGTERM which uses exit() to end cleanly.

Reply:
Yeah, SIGKILL is pretty rare that you'd need it. I was seeing the shutdown function run on SIGTERM / ctrl+c though so it might be different for a process started via the CLI, or for Symfony commands.

The file does get cleared in my case.

Reply:
Yeah, they have some note about commands not being available unless further setup is needed, see: https://wpbrowser.wptestkit.dev/modules/wpcli

Reply:
You'd think it would execute in context of the WP app as you have to provide the `--path` option in the WPCLI module config 🤔 I might try it again with some other commands

Reply:
Cheers for merging the PR with the default WP CLI config, that takes care of using `$I->cli()` etc... but doesn't cover every case still.

We maybe don't want to install the entire wp-cli bundle (at least not until the local tools are all contained under a dev-only dependency) so it's still useful to be able to run `exec( 'wp ...' )` for now.

Cavalcade is a bit of a special one because to run the jobs it calls `wp cavalcade run <id>` and we can't directly control that as it comes from the runner.